presto-accumulo/src/main/java/io/prestosql/plugin/accumulo/AccumuloMetadata.java

```diff
@@ -352,6 +334,42 @@ public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHan
         return columns.build();
     }
 
+    @Override
+    public boolean usesLegacyTableLayouts()
+    {
+        return false;
+    }
+
+    @Override
+    public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(ConnectorSession session, ConnectorTableHandle table, Constraint constraint)
+    {
+        AccumuloTableHandle handle = (AccumuloTableHandle) table;
+
+        TupleDomain<ColumnHandle> oldDomain = handle.getConstraint();
+        TupleDomain<ColumnHandle> newDomain = oldDomain.intersect(constraint.getSummary());
+        if (oldDomain.equals(newDomain)) {
+            return Optional.empty();
+        }
+
+        handle = new AccumuloTableHandle(
+                handle.getSchema(),
+                handle.getTable(),
+                handle.getRowId(),
+                newDomain,
+                getColumnConstraints(handle.getRowId(), newDomain),
```

Comment:
Why do we need to hold on to the column constraints in the table handle? It seems like we should be able to do this directly where they are needed (i.e., in `getSplits`)

Reply:
Column constraints are also similar to `TupleDomain` which would remain same for all splits so thought of moving it to the `AccumuloTableHandle`.

Reply:
As far as I can tell, they are only used in `getSplits()` to call `client.getTabletSplits(...)`, but they are not stored in splits. So we could derive them directly in `getSplits()` from the information in the handle.

Reply:
It was passed to `AccumuloRecordSet` and `AccumuloRecordCursor` but it is not used anywhere expect in getting the split metadata. So we can maintain it as `TupleDomain` and derive it from table handle.